lib/base/benchmark.hpp

```diff
@@ -0,0 +1,37 @@
+/* Icinga 2 | (c) 2024 Icinga GmbH | GPLv2+ */
+
+#pragma once
+
+#include "base/atomic.hpp"
+#include <chrono>
+
+namespace icinga
+{
+
+/**
+ * Benchmark result.
+ *
+ * @ingroup base
+ */
+class Benchmark
+{
+public:
+	using Clock = std::chrono::steady_clock;
+
+	Benchmark& operator+=(const Clock::duration&) noexcept;
+	Benchmark& operator+=(const Clock::time_point&) noexcept;
+
+	/**
+	 * @return The total accumulated time in seconds
+	 */
+	template<class T>
+	explicit operator T() const noexcept
+	{
+		return std::chrono::duration<T>(Clock::duration(m_Sum.load(std::memory_order_relaxed))).count();
```

Comment:
std::memory_order_relaxed is fine here for the same reasons as in https://github.com/Icinga/icinga2/pull/10248#discussion_r1875665859